### PR TITLE
PLAT-1557: Form components invalid state

### PR DESCRIPTION
**Pull request by f-jauregui**
[JIRA ticket](https://reciprocitylabs.atlassian.net/browse/PLAT-1557)

#### Description
As per JIRA ticket

#### ChangeLOG

[Changelog](https://zen-ui.zengrc.com/?path=/docs/changelog--page)

#### Useful links
[Zen UI readme](https://github.com/reciprocity/zen-ui/blob/main/README.md)

[PR review guidelines](https://github.com/reciprocity/zengrc/blob/develop/doc/dev_environment_and_process/pull_request_reviews.md)

[Git commit guidelines](https://github.com/reciprocity/zengrc/blob/develop/doc/dev_environment_and_process/commit_guidelines.md)

### CODE REVIEW

**Review by homer0 — APPROVED** [review]
I just tested the 3 components, and it looks great 🤘

Tiny comment just in case someone notices it too: it may look a little weird-ish around the dropdown (being the host a rectangle and the actual `.field` has border radius). This is fine for now.

File: src/components/zen-dropdown/zen-dropdown.scss [hunk 1]
```diff
@@ -76,3 +76,8 @@ $item-ypadding: 0.5rem;
 .list {
   max-height: #{4 * ($item-line-height + 2 * $item-ypadding)};
 }
+
+:host([invalid='true']) {
```
Comment (chavlji) [inline src/components/zen-dropdown/zen-dropdown.scss:80]:
```suggestion
:host([invalid='true']) .field {
```

Comment (chavlji) [reply src/components/zen-dropdown/zen-dropdown.scss:80]: Agree to @homer0 .

**Review by chavlji — DISMISSED** [review]
Nice.
A few requests:
1. Input's border animates, while dropdown's and textarea's doesn't. Make it consistent.
2. Also implement for date picker

**Comment by f-jauregui** [comment]
> Nice.
> A few requests:
> 
> 1. Input's border animates, while dropdown's and textarea's doesn't. Make it consistent.
> 2. Also implement for date picker

@chavlji 
1. Shall I animate all others or remove animation from input?
2. It is implemented on the date picker (at least the disabled state).

**Review by homer0 — APPROVED** [review]
I just checked both of the things @chavlji requested and they are both implemented and working.

Merge it with :fire:!

**Pull request merged**